### PR TITLE
Don't touch MySQL database if it's unnecessary

### LINKED ISSUE #15706: cannot query system.parts when MySQL server used as database engine is down
**Describe the bug**
Any query to system.parts doesn't work when MySQL server used as database engine not accessible.

**How to reproduce**
* Which ClickHouse server version to use
tested on 20.5.2.7 and 20.9.2.20

```
CREATE DATABASE mysql ENGINE = MySQL('mysql:3306', 'db', 'user', 'password');

SELECT NULL
FROM system.parts
LIMIT 1

┌─NULL─┐
│ ᴺᵁᴸᴸ │
└──────┘

-- no results here
SELECT NULL
FROM system.parts
WHERE database = 'mysql'
LIMIT 1

Ok.

0 rows in set. Elapsed: 0.025 sec.
```
shutdown mysql
```
SELECT NULL
FROM system.parts
LIMIT 1

Received exception from server (version 20.9.2):
Code: 1000. DB::Exception: Received from clickhouse-dev209:9000. DB::Exception: mysqlxx::ConnectionFailed: Unknown MySQL server host 'mysql' (-2) ((nullptr):0).

-- but works here
SELECT NULL
FROM system.parts
WHERE database != 'mysql'
LIMIT 1

┌─NULL─┐
│ ᴺᵁᴸᴸ │
└──────┘
```

**Expected behavior**
SELECT FROM system.parts works without WHERE database NOT IN (_all MySQL dbs_)

### PR DESCRIPTION
**Pull request by Avogar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix bug with MySQL database. When MySQL server used as database engine is down some queries raise Exception, because they try to get tables from disabled server, while it's unnecessary. For example, query `SELECT ... FROM system.parts` should work only with MergeTree tables and don't touch MySQL database at all. 

This closes #15706

### CODE REVIEW

**Review by alexey-milovidov — COMMENTED** [review]
LGTM but a test needed.

**Comment by tavplubix** [comment]
Some queries like `select * from system.tables` or `select * from system.columns` still may fail when MySQL is not available for the same reason. Maybe we should return cached result from `DatabaseConnectionMySQL::getCreateTableQueryImpl(...)` and `DatabaseConnectionMySQL::getTablesIterator(...)`?

**Comment by Avogar** [comment]
> Some queries like `select * from system.tables` or `select * from system.columns` still may fail for the same reason when MySQL is not available. Maybe we should return cached result from `DatabaseConnectionMySQL::getCreateTableQueryImpl(...)` and `DatabaseConnectionMySQL::getTablesIterator(...)`?

In queries like `select * from system.tables` we want to see all tables, even in MySQL databases, I guess it's ok to get exception. This PR is about queries where we don't need to touch MySQL databases at all and we don't expect exception.

File: src/Interpreters/AsynchronousMetrics.cpp [hunk 1]
```diff
@@ -233,8 +233,8 @@ void AsynchronousMetrics::update()
 
         for (const auto & db : databases)
         {
-            /// Lazy database can not contain MergeTree tables
-            if (db.second->getEngineName() == "Lazy")
+            /// Check if database can contain MergeTree tables
+            if (!db.second->canContainMergeTreeTables())
```
Comment (alexey-milovidov) [inline src/Interpreters/AsynchronousMetrics.cpp:237]: This is a bug.

**Pull request merged**